openmetrics/src/main/java/io/airlift/openmetrics/MetricsResource.java

```diff
@@ -246,7 +246,10 @@ private String sanitizeMetricName(String name)
                 .replace(":", "_")
                 .replace("=", "_")
                 .replace(",", "_")
-                .replace("-", "_");
+                .replace("-", "_")
+                .replace(" ", "_")
+                .replace("(", "")
+                .replace(")", "");
```

Comment:
what about having a list of allowed characters instead of a list of characters to replace?
for example, should `%` or `;` be allowed or not?

Reply:
Yes, we should replace characters other than allowed. @vinay-kosaraju can you apply this logic?

Reply:
Will update the PR

Reply:
Updated the PR to replace any non-word character pattern match with `_`. Here's prometheus [metric name convention](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels)

```
\w | A word character: [a-zA-Z_0-9]
-- | --
\W | A non-word character: [^\w]
```

Reply:
A regex pattern might get costly here in cpu. I'd rather stick with a list or precompile a regex. We should only do this for non-jmxutils metrics at least, since those are restricted in the special characters that can come up.

Reply:
Considering Matt's concerns regarding performance, I undid my change. @findepi @wendigo let me know what you think?

Reply:
@vinay-kosaraju you can use `com.google.common.base.CharMatcher` for regex-like character classes without regex-like overhead.

Reply:
```
    // https://github.com/OpenObservability/OpenMetrics/issues/53
    private static final CharMatcher NON_ALLOWED_LABEL_CHARACTERS = CharMatcher
            .inRange('a', 'z')
            .or(CharMatcher.inRange('A', 'Z'))
            .or(CharMatcher.inRange('0', '9'))
            .or(CharMatcher.anyOf("_:"))
            .negate()
            .precomputed();
```

and then:

```
private static String sanitizeMetricName(String name)
    {
        return NON_ALLOWED_LABEL_CHARACTERS.replaceFrom(name, '_');
    }
```

Reply:
Also please add test for that method

Reply:
Thanks! Updated the PR